src/core/thread/mle_router.cpp

```diff
@@ -3715,10 +3715,12 @@ ThreadError MleRouter::AppendActiveDataset(Message &aMessage)
     ThreadError error = kThreadError_None;
     Tlv tlv;
 
+    VerifyOrExit(mNetif.GetActiveDataset().GetNetwork().GetSize() > 0,);
+
     tlv.SetType(Tlv::kActiveDataset);
-    tlv.SetLength(static_cast<uint8_t>(mNetif.GetActiveDataset().GetNetwork().GetSize()));
+    tlv.SetLength(static_cast<uint8_t>(mNetif.GetActiveDataset().GetNetwork().GetSize()) - sizeof(ActiveTimestampTlv));
```

Comment:
It would be cleaner to just have the `Dataset` method append the entire TLV, including the TLV header.